test/arithmetic/multiplier_test.dart

```diff
@@ -198,17 +306,18 @@ void main() {
         } else {
           signedSelect = null;
         }
-        for (final radix in [2, 16]) {
-          for (final width in [5, 6]) {
+        for (final radix in [2, 4]) {
+          for (final width in [3, 4]) {
             for (final ppTree in [KoggeStone.new]) {
               testMultiplyAccumulateRandom(
                   width,
                   10,
                   curryMultiplierAsMultiplyAccumulate(radix, ppTree,
                       ppGen: PartialProductGeneratorStopBitsSignExtension.new,
-                      signed: !signedOperands && signedTest,
-                      selectSign: signedSelect),
-                  signedTest: signedTest);
+                      signedMultiplicand: !signedOperands && signedTest,
+                      signedMultiplier: !signedOperands && signedTest,
+                      selectSignedMultiplicand: signedSelect,
```

Comment:
i see some places where both the multiplicand and multiplier are tied to the same signal -- are you sure you've tested all the combinations of signed and unsigned here?

Reply:
I have the most extensive tests in multiply_encoder_test.dart which does this at the partial product generation level.
But I added complete signage tests at the multiplier level now.  One exhaustive test for all permutations of parameters is too long (signed/unsigned/selected per operand, sign extension methods, final adder generator, widthX, widthY, radix).